### DISCUSSION TITLE
How to avoid Datepicker from triggering form validation too early

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#1659] [Q&A] [answered]
In the video below, you can see that while `react-hook-form` `onBlur` mode looks good for text inputs, it is problematic for the datepicker.

Since the datepicker input field loses focus after clicking it (the focus correctly moves to the actual datepicker modal for a11y reasons etc.) the `onBlur` error validation is triggered before the user had a chance to enter a value.

Is there a way to avoid this? It's unfortunately a business requirement to use `onBlur` rather than `onSubmit` mode.  

https://user-images.githubusercontent.com/16797721/174743157-7cdeb2b0-be6a-4103-99a7-28586205c8cb.mp4

**Comment by rtivital** [comment, marked as answer]
You can set `allowFreeInput` prop, then field will not blur when user picks the date

**Reply by mcmxcdev** [reply]
Not perfect, but works for that use case! Thanks ;)
